staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -104,7 +104,7 @@ func (i *informerSync) Check(_ *http.Request) error {
 	// Close stopCh to force checking if informers are synced now.
 	close(stopCh)
 
-	var informersByStarted map[bool][]string
```

Comment:
this appears to have been an unconditional bug... this needs a testcase that demonstrates the bug so the readyz endpoint doesn't regress

Reply:
This code can only work if it's never called.

Delete the whole function?

Otherwise, yeah, apparently we need some tests for more than just "does it crash".

cc @sttts @logicalhan

Reply:
curious that TestLivezAndReadyz did not catch this

Reply:
Sorry this is my bad. Yes this needs tests. We should stub out the one informer interface function so that we can unit test the checking behavior:

```
type waitForSync interface {
	WaitForCacheSync(stopCh <-chan struct{}) map[reflect.Type]bool
}
```

then we can test that this thing does what we want it to do when it returns various things.

/cc @mborsz @wojtek-t

Reply:
> curious that TestLivezAndReadyz did not catch this

I'm guessing it doesn't get added.

Reply:
@logicalhan , with the latest force-push (https://github.com/kubernetes/kubernetes/commit/b266c34e71f09407b6373daa325f19c90800248d) I added tests that would catch this. 
I am not sure that I get you proposal so that's why I added a stub that allows me to test the func. Please let me know if it is ok.

Reply:
@logicalhan Actually the problem was introduced by this comment: https://github.com/kubernetes/kubernetes/pull/92693 which I blindly applied...